sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,853 @@
+    n_features_out_ : int
+        The dimensionality of a sample's vector after applying to it the linear
+        transformation.
+
+    classes_ : array-like, shape (n_classes,)
+        The appearing class labels.
+
+    n_iter_ : int
+        The number of iterations of the optimizer.
+
+    n_funcalls_ : int
+        The number of times the optimizer computes the loss and the gradient.
+
+    details_ : dict
+        A dictionary of information created by the L-BFGS optimizer during
+        fitting.
+
+
+    Examples
+    --------
+    >>> X = [[0], [1], [2], [3]]
+    >>> y = [0, 0, 1, 1]
+    >>> from sklearn.neighbors import LargeMarginNearestNeighbor
+    >>> lmnn = LargeMarginNearestNeighbor(n_neighbors=1)
+    >>> lmnn.fit(X, y) # doctest: +ELLIPSIS
+    LargeMarginNearestNeighbor(...)
+    >>> print(lmnn.predict([[1.1]]))
+    [0]
+    >>> print(lmnn.predict_proba([[0.9]]))
+    [[ 1.  0.]]
+
+
+    References
+    ----------
+    .. [1] `Weinberger, K. Q.; Saul L. K. (2009). "Distance Metric Learning
+    for Large Margin Classification". Journal of Machine Learning Research.
+    10: 207–244.
+    <http://jmlr.csail.mit.edu/papers/volume10/weinberger09a/weinberger09a.pdf>`_
+
+    .. [2] `Wikipedia entry on Large Margin Nearest Neighbor
+           <https://en.wikipedia.org/wiki/Large_margin_nearest_neighbor>`_
+
+    """
+
+    def __init__(self, L=None, n_neighbors=3, n_features_out=None,
+                 max_iter=200, tol=1e-5, use_pca=True,
+                 max_constraints=int(1e7),
+                 use_sparse=True, warm_start=False,
+                 max_corrections=100, verbose=False, iprint=False,
+                 random_state=None, n_jobs=1):
+
+        super(LargeMarginNearestNeighbor, self).__init__(
+            n_neighbors=n_neighbors, n_jobs=n_jobs)
+
+        # Parameters
+        self.L = L
+        self.n_features_out = n_features_out
+        self.max_iter = max_iter
+        self.tol = tol
+        self.use_pca = use_pca
+        self.max_constraints = max_constraints
+        self.use_sparse = use_sparse
+        self.warm_start = warm_start
+        self.max_corrections = max_corrections
+        self.verbose = verbose
+        self.iprint = iprint
+        self.random_state = random_state
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """Find a linear transformation by optimization of the unconstrained
+        problem, such that the k-nearest neighbor classification accuracy
+        improves.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features_in)
+            The training samples.
+
+        y : array-like, shape (n_samples,)
+            The corresponding training labels.
+
+        Returns
+        -------
+        self : returns a trained LargeMarginNearestNeighbor model.
+
+        """
+
+        # Check inputs consistency
+        X, y_ = self._validate_params(X, y)
+
+        self._random_state = check_random_state(self.random_state)
```

Comment:
I think that you should not store in `self`. You should better pass locally create `random_state` and pass it to the function which need it.